src/NuGet.Core/NuGet.Packaging/Signing/Signer.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    /// <summary>
+    /// Remove or add signature package metadata.
+    /// </summary>
+    public class Signer
+    {
+        // Temporary
+        private const string TestSignedPath = "testsigned/signed.json";
+        private readonly ISignPackage _package;
+
+        /// <summary>
+        /// Creates a signer for a specific package.
+        /// </summary>
+        /// <param name="package">Package to sign or modify.</param>
+        public Signer(ISignPackage package)
+        {
+            _package = package ?? throw new ArgumentNullException(nameof(package));
+        }
+
+        /// <summary>
+        /// Add a signature to a package.
+        /// </summary>
+        public async Task SignAsync(Signature signature, ILogger logger, CancellationToken token)
+        {
+            var signedJson = new JObject();
+            var signatures = new JArray();
+            signedJson.Add(new JProperty("signatures", signatures));
+            var sig = new JObject();
+            signatures.Add(sig);
+            sig.Add(new JProperty("trust", signature.TestTrust.ToString()));
+            sig.Add(new JProperty("type", signature.Type.ToString()));
+            sig.Add(new JProperty("name", signature.DisplayName));
+
+            using (var stream = new MemoryStream())
+            using (var writer = new StreamWriter(stream))
+            {
+                writer.Write(signedJson.ToString());
+                stream.Position = 0;
+                await _package.AddAsync(TestSignedPath, stream, token);
+            }
+        }
+
+        /// <summary>
+        /// Remove all signatures from a package.
+        /// </summary>
+        public Task RemoveSignaturesAsync(ILogger logger, CancellationToken token)
```

Comment:
Consider renaming the class. Signer, does not seem fit for a class that can remove signatures. Maybe `SignUtility`?

Reply:
Operations to overwrite the signature will need both, I think it makes sense they stay together.